pkg/reconciler/buildrun/resources/sources/git.go

```diff
@@ -79,6 +79,15 @@ func AppendGitStep(
 		)
 	}
 
+	// Check if shallow clone is requested
+	if source.Depth != nil && *source.Depth > 0 {
```

Comment:
Don't we need to pass `--depth=0` as well to indicate that the full history should be cloned ?

Reply:
Based on our previous discussion, I understood that we would only specify the depth in the parameter, which would then be attached to git clone --depth. From previous discussions within the community I thought that putting depth as 0 might be confusing for users. Should we adhere to the convention of using 0 to indicate a full history clone? @adambkaplan thoughts?

Reply:
@SaschaSchwarze0 defaulting to 1 makes a lot of sense but putting 0 or -1 for a whole git clone history seems counter-intutive to me. In my opinion either we should just let it be so that it mirrors official git binary behaviour where not giving the depth should indicate a complete history clone or we should put it behind some other boolean parameter.

Reply:
I am not agreeing to that @ayushsatyam146. The majority of users do not understand what a Git history is. They also do not have scenarios where they require the full Git history during their build. Shipright's decision that we clone with depth=1 makes therefore imo sense because it speeds up the source code retrieval for them. Advanced users who understand the concept and have a scanerio where they require the history in their build then have to opt-in to get the full history by specifying depth=0.